Dataset Description: The dataset contains information about various podcast episodes and their attributes. The goal is to analyze and predict the average listening duration of podcast episodes based on various features.

Columns in the Dataset:

Podcast_Name (Type: string)
Description: Names of popular podcasts.
Example Values: "Tech Talk", "Health Hour", "Comedy Central"

Episode_Title (Type: string)
Description: Titles of the podcast episodes.
Example Values: "The Future of AI", "Meditation Tips", "Stand-Up Special"

Episode_Length (Type: float, minutes)
Description: Length of the episode in minutes.
Example Values: 5.0, 10.0, 30.0, 45.0, 60.0, 90.0

Genre (Type: string)
Description: Genre of the podcast episode.
Possible Values: "Technology", "Education", "Comedy", "Health", "True Crime", "Business", "Sports", "Lifestyle", "News", "Music"

Host_Popularity (Type: float, scale 0-100)
Description: A score indicating the popularity of the host.
Example Values: 50.0, 75.0, 90.0

Publication_Day (Type: string)
Description: Day of the week the episode was published.
Possible Values: "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"

Publication_Time (Type: string)
Description: Time of the day the episode was published.
Possible Values: "Morning", "Afternoon", "Evening", "Night"

Guest_Popularity (Type: float, scale 0-100)
Description: A score indicating the popularity of the guest (if any).
Example Values: 20.0, 50.0, 85.0

Number_of_Ads (Type: int)
Description: Number of advertisements within the episode.
Example Values: 0, 1, 2, 3

Episode_Sentiment (Type: string)
Description: Sentiment of the episode's content.
Possible Values: "Positive", "Neutral", "Negative"

Listening_Time (Type: float, minutes)
Description: The actual average listening duration (target variable).
Example Values: 4.5, 8.0, 30.0, 60.0

In [1]:
# import required modules for this project

import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
# Load csv file
df = pd.read_csv('/Users/sa26/Documents/GitHub/Predict-Podcast-Listening-Time/data/raw/train.csv')

# print out the first 5 rows of data
df.head()

id     Podcast_Name Episode_Title  Episode_Length_minutes       Genre  \
0   0  Mystery Matters    Episode 98                     NaN  True Crime   
1   1    Joke Junction    Episode 26                  119.80      Comedy   
2   2   Study Sessions    Episode 16                   73.90   Education   
3   3   Digital Digest    Episode 45                   67.17  Technology   
4   4      Mind & Body    Episode 86                  110.51      Health   

   Host_Popularity_percentage Publication_Day Publication_Time  \
0                       74.81        Thursday            Night   
1                       66.95        Saturday        Afternoon   
2                       69.97         Tuesday          Evening   
3                       57.22          Monday          Morning   
4                       80.07          Monday        Afternoon   

   Guest_Popularity_percentage  Number_of_Ads Episode_Sentiment  \
0                          NaN            0.0          Positive   
1                        75.95            2.0          Negative   
2                         8.97            0.0          Negative   
3                        78.70            2.0          Positive   
4                        58.68            3.0           Neutral   

   Listening_Time_minutes  
0                31.41998  
1                88.01241  
2                44.92531  
3                46.27824  
4                75.61031

From opening the df in Data Wrangler, I am noticing some columns with missing and distinct values. 
'Episode_Length_minutes', 'Guest_Popularity_percentage', and 'Number_of_Ads' have missing values.
Guests might not always be present on a podcast. 

There are 48 distinct podcast names, 100 distinct episode titles. Most frequent podcast name is "Tech Talks" with 22,847 values. Most frequent episode title is "Episode 71" with 10,515 values. 

There are some outliers with episode length and host popularity. The average and median podcast length are a little over one hour. Most podcasts fall within 30 to 90 minutes. There is a podcast episode that lasted over five hours! I would want to investigate the podcast episodes reported to be zero minutes. The median listening length is ~43 minutes, while the mean is ~45. The max is a ~120 minutes! This reveals most listeners don't completely finish a podcast. 

Host popularity mean and median is reported to be ~60%. Most fall between 39 to 80%. The maximum is reported to be almost 120% even though the scale is 0-100. 

Guest popularity mean and median is reported to be ~53%. Most fall between 28 to 77%. There are most likely outliers here since there are values above the max scale of 100. 

Sports, technology, and true crime are among the most popular genres out of ten.

Podcasts are posted everyday with most published on Sunday, Monday, and Friday.

There are 12 distinct number of ads values with some high outliers (over 100 ads!). This column is heavily skewed to the right. Most podcasts have one ad per episode. This column is supposed to be int not float, so some adjustments will be made. 

Podcasts are pretty evenly posted throughout night, evening, afternoon, and morning.

Episode sentiment is prety evenly split between neutral, negative, and positive. 

There are 750,000 rows x 12 columns in the dataset. 